main.py

```
@@ -0,0 +1,79 @@
+from scapy.all import conf, IFACES
+from EthernetPacket import EthernetPacket
+
+
+INTERFACE = "Intel(R) Wi-Fi 6 AX201 160MHz"
+MY_MAC = b'\x78\x2b\x46\x35\x95\x4e'
+
+
+# Print all the available interfaces.
+def print_interfaces() -> None:
+    IFACES.show()
+
+
+"""
+Returns a raw socket given an interface.
+
+@param iface - the given interface.
+
+@return raw socket for the given interface.
+"""
+def create_raw_socket(iface : str) -> any:
+    return conf.L2socket(iface=iface, promisc=True)
+
+
+"""
+Return a string of hex-bytes of the given binary.
+
+@param binary - the given binary.
+
+@return the string of hex-bytes.
+"""
+def to_hex_bytes(binary : bytes) -> str:
+    return ''.join(f'\\x{byte:02x}' for byte in binary)
+
+
+"""
+Return true if the ethernet packet is sent to the defined mac.
+
+@param ethernet_packet - the ethernet packet of type EthernetPacket.
+
+@return True if it meant for us.
+"""
+def is_sent_to_us(ethernet_packet : 'EthernetPacket') -> bool:
+    return MY_MAC == ethernet_packet.getDestinationMAC()
+
+
+"""
+The whole TCP-IP Stack, one layer after another.
+
+@param raw_packet - the raw packet
+
+@return DON'T KNOW YET
+"""
+def tcp_ip_stack(raw_packet : bytes) -> any:
+    #Layer 2
+    try:
+        ethernet_packet : 'EthernetPacket' = EthernetPacket(raw_packet)
+
+        if not is_sent_to_us(ethernet_packet):
+            print("The ethernet packet is not meant for us!")
+            return
+
+        print("The ethernet packet is meant for us! Moving on...")
+        #NEXT LEVELS
+    
+    except Exception as e:
```

It is a very bad practice to catch all exceptions.
If you are not expecting an exception don't try to catch it.
`try... except` blocks do not make errors magically go away, they just hide them from you when you try to debug them